processing/src/main/java/org/apache/druid/query/topn/TopNQueryEngine.java

```diff
@@ -275,7 +275,7 @@ private static boolean canUsePooledAlgorithm(
       final int numBytesToWorkWith = resultsBuf.capacity();
       final int numValuesPerPass = numBytesPerRecord > 0 ? numBytesToWorkWith / numBytesPerRecord : cardinality;
 
-      return numValuesPerPass <= cardinality;
+      return numValuesPerPass >= cardinality;
```

Comment:
this comparision change caught my eye...the new cmp seem to be more in line with the intention - but it seems to me that
depending on some conditions [here](https://github.com/apache/druid/blob/24e5d8a9e81fffd656e9544ae16bea53d7e60808/processing/src/main/java/org/apache/druid/query/topn/TopNQueryEngine.java#L111-L115) ; this `cardinality`'s value might be `-1` in some cases
is that ok to answer `true` if `cardinality` is `-1` ?

note: if `cardinality == -1` ; then both the old and the new logic returns `true` - is that ok?

Reply:
yea this is fair, it is not really a problem in practice because currently the column capabilities should not report as dictionary encoded if the cardinality is -1, so we return false earlier in the method before we get here, but this seems worth explicitly checking for just in case.